### PR TITLE
Fail fast when a mutation-CLI flag is given no value

### PR DESCRIPTION
**Pull request by stefan-burke**
## What this changes

Running `deno task mutation --source` (or `--test`, `--jobs`, `--timeout`) with nothing after the flag used to quietly treat the flag itself as a positional value. That mistake then surfaced much later as a confusing message like "No source files matched" or "Unexpected positional argument", pointing the person at the wrong thing.

Now the command stops straight away with a clear message: **"Missing value for --source."**

## How

- The pure command-line parsing has moved out of the CLI script into a new, self-contained module `scripts/mutation/args.ts`. `scripts/mutation.ts` is now a thin shell that expands globs and runs the tests; the parsing has no file access, so it can be tested on its own.
- `applyArg` now recognises a value flag with no following token as a usage error instead of silently collecting the flag as a positional argument.

## Tests

New `test/scripts/mutation-args.test.ts` covers `parseArgs` with 100% line and branch coverage and a 100% mutation-kill rate, including the missing-value regression for all four value flags and the non-negative/positive boundaries for `--timeout` and `--jobs`.

## Origin

`TODO.md` → "Code-quality detector & test-strengthening follow-ups (from PR #1729)", item 3. The TODO entry is removed as part of this change.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

https://claude.ai/code/session_0191NXZLeokw3pYtJCwJK4VN

---
_Generated by [Claude Code](https://claude.ai/code/session_0191NXZLeokw3pYtJCwJK4VN)_

## Summary by CodeRabbit

* **Bug Fixes**
  * Improved `mutation` command-line parsing with stricter handling of boolean flags, value flags, and positionals.
  * Ensures `--jobs` and `--timeout` are validated reliably (including missing values, non-numeric input, and invalid ranges), with consistent error precedence.
  * Clarified how source/test values are interpreted when provided as positionals versus `--source`/`--test` flags (including edge cases).
* **Tests**
  * Added comprehensive coverage for defaults, repeated options, positional parsing rules, missing-value errors, numeric validation, and precedence regressions.

### CODE REVIEW

File: scripts/mutation/args.ts [hunk 1]
```diff
@@ -0,0 +1,126 @@
...
+/** Options that take a value; maps the flag to how it records its value. */
+const VALUE_FLAGS: Record<string, ValueFlagApply | undefined> = {
+  "--jobs": (parsed, value) => {
+    parsed.batchJobs = Number(value);
+  },
+  "--source": (parsed, value) => parsed.sources.push(value),
+  "--test": (parsed, value) => parsed.tests.push(value),
+  "--timeout": (parsed, value) => {
+    parsed.timeout = Number(value);
+  },
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply scripts/mutation/args.ts]: Fixed in e0e9613. Numeric flag values now go through a `toNumber` helper that maps a blank/whitespace-only token to `NaN` (instead of `Number("")`/`Number(" ")` → 0), so `--timeout ""` and `--jobs " "` are rejected by the existing numeric validation with a clear message. Both numeric flags share it. Regression tests: "rejects an empty --timeout value instead of reading it as zero" and "rejects a whitespace-only --jobs value".

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/mutation/args.ts [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+): boolean => {
+  if (arg === "--exhaustive") parsed.exhaustive = true;
+  else if (arg === "--harness") parsed.useHarness = true;
+  else if (arg === "-h" || arg === "--help") parsed.help = true;
+  else {
+    const applyValue = VALUE_FLAGS[arg];
+    if (applyValue === undefined) positional.push(arg);
+    else if (next === undefined) parsed.error = `Missing value for ${arg}.`;
+    else {
+      applyValue(parsed, next);
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply scripts/mutation/args.ts]: Fixed in e0e9613. `VALUE_FLAGS` (and the new `BOOLEAN_FLAGS`) are now `Map`s rather than object literals, so a lookup can never resolve an inherited `Object.prototype` member — `__proto__`/`constructor`/`toString` are treated as ordinary positionals and nothing is invoked as a handler. Regression test: a table "treats the prototype-named token … as a positional" over `__proto__`, `constructor`, and `toString`.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/mutation/args.ts [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+      return true;
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply scripts/mutation/args.ts]: Fixed in e0e9613. A new `isKnownFlag` predicate — derived from the same `BOOLEAN_FLAGS`/`VALUE_FLAGS` maps, so it can't drift — makes `applyArg` treat a following *known option* as a missing value: `--source --test t.ts` now reports "Missing value for --source." rather than swallowing `--test`. The first error is preserved via `??=`. Negative numeric values still work — `-5` isn't a known flag, so `--timeout -5` is still consumed and rejected as invalid. Regression tests: "treats a following value flag as a missing value", "treats a following boolean flag as a missing value", and "keeps the first missing-value error when two flags both lack a value".

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/mutation.ts [hunk 4]
```diff
@@ -18,6 +18,7 @@
  */
 
 import { globToRegExp, join, normalize, SEPARATOR } from "@std/path";
+import { DEFAULT_TIMEOUT, parseArgs } from "./mutation/args.ts";
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply scripts/mutation.ts:21]: Not actionable — this looks like a diff-reading artifact. `USAGE` is still defined locally in `scripts/mutation.ts` (lines 31–49) and `main()`'s `console.log(USAGE)` resolves against it; only the parsing helpers moved to `args.ts`. `USAGE` stayed and now interpolates the imported `DEFAULT_TIMEOUT`. The line removed just above `USAGE` was `const DEFAULT_TIMEOUT = 10_000;`, which likely made the hunk look like `USAGE` went with it. `deno task typecheck` (and the full `deno task precommit`, which typechecks test files too) pass on the current commit.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/mutation/args.ts [hunk 5]
```diff
@@ -0,0 +1,126 @@
...
+
+export interface ParsedArgs {
+  batchJobs?: number;
+  error: string | null;
+  exhaustive: boolean;
+  help: boolean;
+  sources: string[];
+  tests: string[];
+  timeout: number;
+  useHarness: boolean;
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply scripts/mutation/args.ts:19]: Not changing this. In this codebase the `*OrNull` convention applies to lookup functions/returns and locals (`decryptAttendeeOrNull`, `firstRowOrNull`), not to struct fields — nullable interface fields use a plain name plus a `| null` type, which already documents the contract. Direct precedent: `siteVersionError: string | null` in `src/shared/site-update.ts` (an error field, same shape), plus many others in `src/shared/types.ts` and `src/shared/merge/attendee-merge-types.ts`. Renaming `error` → `errorOrNull` would make this field inconsistent with every other nullable field in the codebase.

---
_Generated by [Claude Code](https://claude.ai/code)_

File: scripts/mutation/args.ts [hunk 6]
```diff
@@ -0,0 +1,126 @@
...
+    }
+    return;
+  }
+  if (positional[0] !== undefined) parsed.sources.push(positional[0]);
+  if (positional[1] !== undefined) parsed.tests.push(positional[1]);
+  if (positional.length > 2) {
+    parsed.error =
+      `Too many positional arguments (${positional.length}). Quote your globs ` +
+      `so the shell can't expand them — e.g. 'src/lib/forms/*.ts' ` +
+      `'test/lib/forms/*.test.ts' — or pass repeated --source/--test flags.`;
```
[the remark this answers is not in the document]
Comment (stefan-burke) [reply scripts/mutation/args.ts:137]: Fixed in e0e9613. Both `applyPositionals` assignments now use `??=`, so a missing-value error recorded earlier by `applyArg` stays authoritative over the stray-positional and too-many-positional messages. Regression tests: "keeps a missing-value error over a stray flag-form positional" (using your exact example `["a.ts", "--source", "src.ts", "--jobs"]` → "Missing value for --jobs.") and "keeps a missing-value error over a too-many-positionals error".

---
_Generated by [Claude Code](https://claude.ai/code)_
